library/ssl_tls13_client.c

```diff
@@ -1709,6 +1706,27 @@ static int ssl_tls13_postprocess_server_hello( mbedtls_ssl_context *ssl )
             goto cleanup;
     }
 
+    MBEDTLS_SSL_DEBUG_MSG( 3,
+            ( "Server selected key exchange mode: %s",
+              handshake->key_exchange_mode ==
+              MBEDTLS_SSL_TLS1_3_KEY_EXCHANGE_MODE_PSK ? "psk" :
+              (handshake->key_exchange_mode ==
+              MBEDTLS_SSL_TLS1_3_KEY_EXCHANGE_MODE_EPHEMERAL ? "ephemeral" :
+              "psk_ephemeral")) );
+
+    if( !mbedtls_ssl_conf_tls13_check_kex_modes( ssl, handshake->key_exchange_mode ) )
+    {
+        ret = MBEDTLS_ERR_SSL_HANDSHAKE_FAILURE;
+        MBEDTLS_SSL_DEBUG_MSG( 2,
+                ( "Not supported kex mode in client: %s",
+              handshake->key_exchange_mode ==
+              MBEDTLS_SSL_TLS1_3_KEY_EXCHANGE_MODE_PSK ? "psk" :
+              (handshake->key_exchange_mode ==
+              MBEDTLS_SSL_TLS1_3_KEY_EXCHANGE_MODE_EPHEMERAL ? "ephemeral" :
+              "psk_ephemeral")) );
+        goto cleanup;
+    }
+
```

Comment:
Improve the readability and adjust the order. 
```suggestion
   
    if( !mbedtls_ssl_conf_tls13_check_kex_modes( ssl, handshake->key_exchange_mode ) )
    {
        ret = MBEDTLS_ERR_SSL_HANDSHAKE_FAILURE;
        MBEDTLS_SSL_DEBUG_MSG( 2, ( "Key exchange mode(%s) is not configured supported."), ssl_get_kex_mode_str( handshake->key_exchange_mode ) );
        goto cleanup;
    }
    
    MBEDTLS_SSL_DEBUG_MSG( 2, ( "key exchange mode: %s", ssl_get_kex_mode_str( handshake->key_exchange_mode ) );
```